plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/EqualityDeleteFilter.java

```diff
@@ -47,22 +45,27 @@ public RowPredicate createPredicate(List<IcebergColumnHandle> columns)
                 .toArray(Type[]::new);
 
         Schema fileSchema = schemaFromHandles(columns);
-        StructProjection projection = StructProjection.create(fileSchema, schema);
+        equalityDeleteSets.initialize(fileSchema);
 
         return (page, position) -> {
             StructLike row = new LazyTrinoRow(types, page, position);
-            return !deleteSet.contains(projection.wrap(row));
+            return !equalityDeleteSets.contains(row);
```

Comment:
I'm having a harder time following this part of the changes. Why is it better to do the projection separately in `EqualityDeleteSet`?

Reply:
Because creating a Projection requires the schema of the Delete file, I moved this part of the implementation to the EqualityDeleteSets, and I also moved the `Projection.wrap` logic to it
```java
projection=StructProjection.create(fileSchema, schema)
```